src/main/utils/RespondentUtils.ts

```diff
@@ -14,4 +19,42 @@ export default class RespondentUtils {
     }
     return request.session.userCase.respondents[request.session.selectedRespondentIndex];
   }
+
+  /**
+   * Determines whether the currently selected respondent is represented.
+   *
+   * This method checks the current request's session to identify if a representative
+   * is assigned to the selected respondent. It performs the following steps:
+   * 1. Verifies that there is at least one representative in the user case.
+   * 2. Retrieves the respondent selected in the request context.
+   * 3. Iterates over the list of representatives and checks if any of them
+   *    are associated with the respondent via matching `respondentId`.
+   *
+   * @param req - The current HTTP request object, containing the session and user case context.
+   * @returns `true` if the selected respondent has a matching representative; otherwise, `false`.
+   */
+  public static respondentRepresented(req: AppRequest): boolean {
```

Comment:
I think there's a slight flaw in this logic
- If there are 4 respondents on a case (A, B, C and D to keep things simple) and they are submitted in this order. On submission, SOLICITORA corresponds to A, SOLICITORB to B and so on according to the NoC answers.
- However, the ordering of respondents can change. If respondent A is struck out, the ordering then looks like B, C, D and then A. Then index of B is now 0 which will point to SOLICITORA but should be SOLICITORB
- I think you'll need to use the noticeOfChangeAnswers0 ->10 to dictate which respondent correlates to the rep role

Reply:
Harpreet, Unfortunately not on all cases we are able to get noticeOfChangeAnswers. As we talked only the online submissions.
So for now I will implement to forward the page to make a request to remove respondent representative. And we need to think about that later.

Reply:
Forwarded to Contact Tribunal whenever user selects "
Yes, I confirm I wish to remove my legal representative and continue my case representing myself."